src/PublisherMethods.js

```diff
@@ -103,5 +105,37 @@ module.exports = {
         _.nextTick(function() {
             me.trigger.apply(me, args);
         });
-    }
+    },
+
+    /**
+     * Returns a Promise for the triggered action
+     */
+    triggerPromise: function(){
+        var me = this;
+        var args = arguments;
+
+        var canHandlePromise =
+            this.children.indexOf('completed') >= 0 &&
+            this.children.indexOf('failed') >= 0;
+
+        if (!canHandlePromise){
+            throw new Error('Publisher must have "completed" and "failed" child publishers');
+        }
+
+        var promise = new Promise(function(resolve, reject) {
```

Comment:
`bluebird` [does have a constructor][1] alongside many factory methods, though `Q` does not, as you noted.

So, with 87c4be3, you could change the Promise constructor.  Your recommendation, however, is letting users set the factory method (which is currently `utils.createPromise`) rather than relying on the new Promise to be a constructor?  This would effectively become an API shortcut to override `utils.createPromise`, correct?

That's easy enough to do...